system/security/plugins/jwtSecurity/jwtCache.hpp

```diff
@@ -0,0 +1,459 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef JWTCACHE_HPP_
+#define JWTCACHE_HPP_
+
+#include <map>
+#include <string>
+
+#include "basesecurity.hpp"
+
+/**
+ * Instances of JWTUserInfo represent all of the cached values for a particular
+ * user.  The intention is that instances would be accessed via another
+ * container, such as a map keyed off the user's username.  See JWTUserCache class.
+ */
+class JWTUserInfo
```

Comment:
Does that apply when a copy of the retrieved value is returned?  I'm using copies, rather than references, to avoid issues with multi-threaded management.

Reply:
The "get" prefix usually suggests that the caller is responsible for releasing the returned value, while "query" usually means you are just looking at somebody else's reference and are not expected to release it. A copy that the caller is responsible for disposing should be "get".